fMRIVolume/scripts/OneStepResampling.sh

```diff
@@ -181,7 +210,12 @@ for ((k=0; k < $NumFrames; k++)); do
   prevmatrix="${MotionMatrixFolder}/${MotionMatrixPrefix}${vnum}"
 
   # Combine GCD with motion correction
-  ${FSLDIR}/bin/convertwarp --relout --rel --ref=${WD}/prevols/vol${vnum}.nii.gz --warp1=${GradientDistortionField} --postmat=${MotionMatrixFolder}/${MotionMatrixPrefix}${vnum} --out=${MotionMatrixFolder}/${MotionMatrixPrefix}${vnum}_gdc_warp.nii.gz
+  if [ "$fMRIReferenceReg" == "nonlinear" ]; then
+      ${FSLDIR}/bin/convertwarp --relout --rel --ref=${WD}/prevols/vol${vnum}.nii.gz --warp1=${GradientDistortionField} --midmat=${MotionMatrixFolder}/${MotionMatrixPrefix}${vnum} --warp2=${MotionMatrixFolder}/postmc2fmriref_warp --out=${MotionMatrixFolder}/${MotionMatrixPrefix}${vnum}_gdc_warp.nii.gz
```

Comment:
Add comment that the name of the nonlinear warp field is also hard-coded into MotionCorrection.sh

Reply:
done.